File: src/Python/101-200/103.BinaryTreeZigzagLevelOrder.py

class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def zigzagLevelOrder(self, root: TreeNode) -> list:
        if root is None:
            return []
        ans = []
        queue = [root]
        order = 1
        while len(queue) > 0:
            tmp = []
            nextqueue = []
            for node in queue:
                tmp.append(node.val)
                if node.left is not None:
                    nextqueue.append(node.left)
                if node.right is not None:
                    nextqueue.append(node.right)
            ans.append(tmp[::order])
            queue = nextqueue
            order = -order
        return ans

if __name__ == "__main__":
    solution = Solution()
    node = TreeNode(1)
    node.left = TreeNode(2)
    node.right = TreeNode(3)
    node.left.left = TreeNode(4)
    node.right.right = TreeNode(5)
    print(solution.zigzagLevelOrder(node))
   

File: src/CSharp/101-200/103.BinaryTreeZigzagLevelOrder.cs

using System;
using System.Linq;
using System.Text;
using System.Collections.Generic;


namespace PeefyLeetCode.BinaryTreeZigzagLevelOrderTraversal
{

public class TreeNode {
    public int val;
    public TreeNode left;
    public TreeNode right;
    public TreeNode(int x) { val = x; }
}

public class Solution {
    public IList<IList<int>> ZigzagLevelOrder(TreeNode root) {
        IList<IList<int>> ans = new List<IList<int>>();
        if (root == null)
            return ans;
        List<TreeNode> level = new List<TreeNode>();
        level.Add(root);
        int order = 1;
        while (level.Count > 0){
            List<int> cur = new List<int>();
            List<TreeNode> next_level = new List<TreeNode>();
            foreach (TreeNode node in level) {
                cur.Add(node.val);
                if (node.left != null)
                    next_level.Add(node.left);
                if (node.right != null)
                    next_level.Add(node.right);
            }
            if (order == -1)
                cur.Reverse();
            ans.Add(cur);
            level = next_level;
            order = -order;
        }
        return ans;
    }
}
}